app/src/main/java/net/squanchy/schedule/SchedulePageView.kt

```diff
@@ -120,25 +124,51 @@ class SchedulePageView @JvmOverloads constructor(
     }
 
     fun updateWith(schedule: Schedule, onEventClicked: (Event) -> Unit) {
-        viewPagerAdapter.updateWith(schedule.pages, onEventClicked)
+        val initialEventForPage = schedule.pages.map(::findNextEventForPage).toTypedArray()
+        viewPagerAdapter.updateWith(schedule.pages, initialEventForPage, onEventClicked)
+
+        val todayPageIndex = findTodayIndexOrDefault(schedule.pages)
+        viewpager.setCurrentItem(todayPageIndex, false)
+
+        tabstrip.addOnTabSelectedListener(ScrollingOnTabSelectedListener(schedule, viewPagerAdapter))
         progressbar.visibility = View.GONE
     }
 
+    private fun findTodayIndexOrDefault(pages: List<SchedulePage>): Int {
+        val now = currentTime.currentLocalDateTime()
+        return pages.firstOrNull { it.date.toLocalDate().isEqual(now.toLocalDate()) }
+            ?.let(pages::indexOf) ?: 0
+    }
+
+    private fun findNextEventForPage(page: SchedulePage) =
+        page
+            .events
+            .firstOrNull {
+                it.startTime.toDateTime(it.timeZone).isAfter(currentTime.currentLocalDateTime().toDateTime(it.timeZone))
+            }
+
+    private inner class ScrollingOnTabSelectedListener constructor(
+            private val schedule: Schedule,
+            private val viewPagerAdapter: ScheduleViewPagerAdapter
+    ) : OnTabSelectedListener {
+        override fun onTabReselected(tab: TabLayout.Tab) {
```

Comment:
Can we have an empty line after the class declaration? Don't know if ktlint caught this, but it's something we have been doing as a project convention